app/vite.config.js

```diff
@@ -6,10 +6,57 @@ import { defineConfig } from "vite";
 import { VitePWA } from "vite-plugin-pwa";
 import { viteStaticCopy } from "vite-plugin-static-copy";
 
-// import path from "path" causes eslint to crash for some reason
+// import from "path" and "fs" causes eslint to crash for some reason
 const path = require("path");
+const {
+  copyFileSync,
+  existsSync,
+  mkdirSync,
+  readFileSync,
+  rmSync,
+  writeFileSync,
+} = require("fs");
 
-export default defineConfig({
+const TEMP_PROD_HTML_DIR = "tempHtmlFiles";
+
+/**
+ *
+ * @param {"build" | "serve"} command
+ * @returns {import("vite").PluginOption}
+ */
+function prodScriptPlugin(command) {
+  return {
+    name: "prod-script",
+    buildStart(options) {
+      if (command === "build") {
+        const htmlFiles = Object.values(options.input).filter((file) =>
+          file.endsWith(".html"),
+        );
+
+        for (const htmlFilePath of htmlFiles) {
+          const html = readFileSync(htmlFilePath, "utf-8");
+
+          const updatedHtml = html
+            // remove vite client import
+            .replace(
+              /<script type="module">\s*import "http:\/\/localhost:5173\/@vite\/client";[\s\S]+?window\.process = \{ env: \{ NODE_ENV: "development" \} \};\s*<\/script>/,
```

Comment:
```suggestion
              /<script type="module">\s*import "http:\/\/localhost:5173\/@vite\/client";[\s\S]+?<\/script>/,
```
Do we need to match the `window.process = {env: {NODE_ENV: "development"}};` part? I'm thinking `import http://localhost:5173/@vite/client;` is unique enough to this block such that we just need to match that. The concern is variable whitespace again: `window.process = {env: {NODE_ENV: "development"}};` does not match `window.process = { env: { NODE_ENV: "development" } };`

Reply:
i tried your regex and it didn't work because of the surrounding scripts, couldn't make it work so just made this one lol. i guess this line is not supposed to be changed so thats fine?

Reply:
yeah i realized now [the regex i originally suggested](https://github.com/igneous-labs/htmx-axum-template/pull/1#discussion_r1308223614) wouldnt work because the `[\s\S]+` before the `</script>` is [greedy](https://www.javascripttutorial.net/javascript-regex-greedy/), so any `<script></script>` blocks that immediately followed would be matched as well. However I think it should work now with `[\s\S]+?` since the additional `?` makes it [non-greedy](https://www.javascripttutorial.net/regular-expression-non-greedy/) so that it stops at the first `</script>`